### PR TITLE
Add workflow to trigger docs site rebuild

### PR DESCRIPTION
**Pull request by echohtp**
Add workflow that sends a `dispatch_event` to `holaplex/marketplace-api-docs` which causes a site rebuild and refresh with the new API details.

### CODE REVIEW

**Review by mpwsh — CHANGES_REQUESTED** [review]
yaml police 🚓  - indentation is off @ line 10

**Review by mpwsh — APPROVED** [review]

File: .github/workflows/dispatch_to_docs.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Send event to rebuild docs
+
+on:
+  push:
+    branches:
+      - main
```
Comment (ray-kast) [inline .github/workflows/dispatch_to_docs.yml:6]:
```suggestion
      - [dev, master]
```

**Review by ray-kast — CHANGES_REQUESTED** [review]

File: .github/workflows/dispatch_to_docs.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Send event to rebuild docs
+
+on:
+  push:
+    branches:
+      - [dev, master]
```
Comment (ray-kast) [inline .github/workflows/dispatch_to_docs.yml:6]:
```suggestion
    branches: [dev, master]
```

**Review by ray-kast — CHANGES_REQUESTED** [review]
My bad.

**Review by ray-kast — APPROVED** [review]

**Pull request merged**